src/main/java/frc/robot/vision/cameras/limelight/LimelightFilters.java

```diff
@@ -17,12 +18,45 @@ public static Filter detectedObjectFilter(Limelight limelight) {
 		return Filter.nonFilteringFilter();
 	}
 
-	public static Filter megaTag1Filter(Limelight limelight) {
-		return Filter.nonFilteringFilter();
+	public static Filter megaTag1Filter(
+		Limelight limelight,
+		Function<Double, Optional<Rotation2d>> wantedYawAtTimestamp,
+		Supplier<Boolean> isYawCalibrated,
+		Translation2d robotInFieldTolerance,
+		Rotation2d yawAtAngleTolerance
+	) {
+		double timestamp = limelight.getMT1RawData().timestampSeconds();
+		return MegaTagFilters.isRobotInField(() -> limelight.getMT1RawData().pose().getTranslation(), robotInFieldTolerance)
+			.and(MegaTagFilters.doesYawExistAtTimestamp(timestamp, wantedYawAtTimestamp))
+			.and(
+				MegaTagFilters.isYawAtExpectedAngle(
+					() -> limelight.getMT1RawData().pose().getRotation(),
+					() -> wantedYawAtTimestamp.apply(timestamp).get(),
+					isYawCalibrated,
+					yawAtAngleTolerance
+				)
+			);
```

Comment:
```suggestion
			.and(
			     MegaTagFilters.doesYawExistAtTimestamp(timestamp, wantedYawAtTimestamp).and(
				     MegaTagFilters.isYawAtExpectedAngle(
					     () -> limelight.getMT1RawData().pose().getRotation(),
					     () -> wantedYawAtTimestamp.apply(timestamp).get(),
					     isYawCalibrated,
					     yawAtAngleTolerance
				     )
			     )
			);
```
to emphasize the filters are dependant